lib/api/kuzzle.js

```diff
@@ -234,6 +240,34 @@ class Kuzzle extends EventEmitter {
       });
   }
 
+   /** Manage errors to be thrown
+   *
+   * @param {...args} ...args
+   * @throws {KuzzleError}
+   */
+
+  throw(...args) {
+
+    const [domain, subdomain, errorName] = args;
+    const placeholders = args.length > 3 ? args.slice(3) : ' ';
+    const name = `${domain}-${subdomain}-${errorName}`;
+    if (!errorCodes.hasOwnProperty(domain)
+      || !errorCodes[domain].subdomains.hasOwnProperty(subdomain)
+      || !errorCodes[domain].subdomains[subdomain].errors.hasOwnProperty(errorName)
+    ) {
+      const message = format('Internal Error : Cannot find error in config file. %s', ...placeholders);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "message" hides or potentially hides a variable declared in an outer scope at line 266. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)